lib/compat.php

```diff
@@ -233,3 +233,25 @@ function gutenberg_render_block_with_assigned_block_context( $pre_render, $parse
 	return $block->render();
 }
 add_filter( 'pre_render_block', 'gutenberg_render_block_with_assigned_block_context', 9, 2 );
+
+/**
+ * Appends the "Uncategorized" block category to the default set of block
+ * categories.
+ *
+ * This can be removed when plugin support requires WordPress 5.5.0+.
+ *
+ * @see Trac Ticket TBD
+ *
+ * @param array $categories Block categories.
+ *
+ * @return array Block categories with "Uncategorized" block category.
+ */
+function gutenberg_append_default_block_category( $categories ) {
+	$categories[] = array(
+		'slug'  => 'uncategorized',
+		'title' => __( 'Uncategorized' ),
+	);
+
+	return $categories;
```

Comment:
What's the reasoning for actually registering a default category instead of just having a nullable category and just group blocks with a "null" category together in the inserter? It might be simpler and avoids having to define a "default category"?

Reply:
> What's the reasoning for actually registering a default category instead of just having a nullable category and just group blocks with a "null" or inexistant category together in the inserter? It might be simpler and avoids having to define a "default category"?

This is mentioned in the original comment:

>- "Uncategorized" may not need to be an actual block category, but instead handled in user interfaces where absence of category is handled.
>   - This may actually be good! I'm only now considering it as I write the pull request comment. I do foresee a couple minor downsides.
>   - Downside: It becomes a risk for duplication and syncing issues if each handling of the "Uncategorized" behavior needs to behave consistently. For example, both the Block Inserter and Block Manager would separately need to implement a (hopefully consistent) "Uncategorized" label. "Hoping" is always a risk, and we could avoid it by treating the default category as a specially-handled case.

It may also require a bit more "explicit" handling in the components where the groupings are arranged, since it wouldn't be able to be treated just the same as any other category.

In the end though, it still may be the simplest option here 🤷 Worth exploring at least.

Reply:
> What's the reasoning for actually registering a default category instead of just having a nullable category and just group blocks with a "null" or inexistant category together in the inserter? It might be simpler and avoids having to define a "default category"?

Still in progress, but mostly completed, and seems quite a bit simpler indeed: #22280